app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,41 @@
+class VotesController < ApplicationController
+  def upvote
+    found = Votes.find_by(user: params[:user], api: params[:api_id])
+
+    if found == nil
+      Votes.create(
+        user: User.find(params[:user]),
+        api: Api.find(params[:api_id]),
+        user_vote: true)
+      render json: Votes.calculate_votes(params[:api_id])
+      return
```

Comment:
What's going on with this `return` without any data passed to it? Why not simply render json at the end of a given logic branch?